library/std/src/net/hostname.rs

```diff
@@ -0,0 +1,10 @@
+use crate::ffi::OsString;
+
+/// Returns the system hostname.
+///
+/// The only time this should ever error out is if the system is broken
+/// in some way. Such errors depend on those cases.
+#[unstable(feature = "gethostname", issue = "135142")]
+pub fn hostname() -> crate::io::Result<OsString> {
```

Comment:
So, funny thing - DNS hostnames can only be valid ASCII. I feel like OsString is overkill here. Would it be appropriate to use std::ascii::Char, if it were available? Or maybe just String?

Reply:
@joshtriplett @valaphee as the main two who were involved in that discussion, could yall maybe provide some insight? 

Would it be an issue to merge it as-is with OsString, and change it to Box<[ascii::Char]> down the line?

Reply:
Oops, didn't see your comment @Noratrieb. Sorry, my browser didn't update the page. Alright, I'll leave it.